class.jetpack.php

```diff
@@ -574,8 +574,7 @@ private function __construct() {
 			};
 		} );
 
-		$this->connection_manager = new Connection_Manager();
-		$this->connection_manager->init();
+		add_action( 'plugins_loaded', array( $this, 'init_connection_manager' ) );
```

Comment:
@roccotripaldi I'm not sure I agree with this change. There are a lot of pieces in the code that rely that `Jetpack::connection()` (respectively `Jetpack::init()`) will grant us with a connection manager instance that we can use immediately. With this change, we will no longer be able to confirm this assumption, and some places will start trying to access an object that's not constructed yet.

What was the motivation for this change? Could we figure out any alternatives?

Reply:
Reverted it, and filed a fix for the bug Rocco discovered in https://github.com/Automattic/jetpack/pull/13270/commits/43722205570a79e7c30b66c7dd3bf7552d739cde.